gcloud-java-dns/src/main/java/com/google/gcloud/dns/ManagedZoneInfo.java

```diff
@@ -0,0 +1,334 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class is a container for the managed zone metainformation. Managed zone is a resource that
+ * represents a DNS zone hosted by the Cloud DNS service. See <a href="https://cloud.google.com/dns/api/v1/managedZones">Google
+ * Cloud DNS documentation</a> for more information.
+ */
+public class ManagedZoneInfo implements Serializable {
+
+  private static final long serialVersionUID = 201601191647L;
+  private final String name;
+  private final Long id;
+  private final Long creationTimeMillis;
+  private final String dnsName;
+  private final String description;
+  private final String nameServerSet;
+  private final List<String> nameServers;
+
+  /**
+   * A builder for {@code ManagedZoneInfo}.
+   */
+  public static class Builder {
+    private String name;
+    private Long id;
+    private Long creationTimeMillis;
+    private String dnsName;
+    private String description;
+    private String nameServerSet;
+    private List<String> nameServers = new LinkedList<>();
+
+    private Builder() {
+    }
+
+    private Builder(Long id) {
+      this.id = checkNotNull(id);
+    }
+
+    private Builder(String name) {
+      this.name = checkNotNull(name);
+    }
+
+    private Builder(String name, Long id) {
+      this.name = checkNotNull(name);
+      this.id = checkNotNull(id);
+    }
+
+    /**
+     * Creates a builder from an existing ManagedZoneInfo object.
+     */
+    Builder(ManagedZoneInfo info) {
+      this.name = info.name;
+      this.id = info.id;
+      this.creationTimeMillis = info.creationTimeMillis;
+      this.dnsName = info.dnsName;
+      this.description = info.description;
+      this.nameServerSet = info.nameServerSet;
+      this.nameServers.addAll(info.nameServers);
+    }
+
+    /**
+     * Sets a mandatory user-provided name for the zone. It must be unique within the project.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets an id for the managed zone which is assigned to the managed zone by the server.
+     */
+    public Builder id(long id) {
+      this.id = id;
+      return this;
+    }
+
+    /**
+     * Sets the time when this managed zone was created.
+     */
+    Builder creationTimeMillis(long creationTimeMillis) {
+      checkArgument(creationTimeMillis >= 0, "The timestamp cannot be negative.");
```

Comment:
Because this is package scope (and basically comes from the proto, I would not have the check).